pkg/template/static_context.go

```diff
@@ -390,14 +395,99 @@ func (ctx StaticCtx) tlsKey(certName string, args ...interface{}) string {
 	return p.Key
 }
 
+func (ctx StaticCtx) tlsCaCert(caName string, args ...interface{}) string {
+	cap, ok := caMap[caName]
+	if !ok {
+		if len(args) != 1 {
+			return ""
+		}
+
+		daysValid, ok := args[0].(int)
+		if !ok {
+			return ""
+		}
+
+		cap = genCa(caName, daysValid)
+		caMap[caName] = cap
+	}
+
+	return cap.Cert
+}
+
+func (ctx StaticCtx) tlsCertFromCa(caName, certName, cn string, ips []interface{}, alternateDNS []interface{}, daysValid int) string {
+	key := fmt.Sprintf("%s:%s:%s", caName, certName, cn)
+	if p, ok := tlsMap[key]; ok {
+		return p.Cert
+	}
+
+	p := genSignedCert(caName, cn, ips, alternateDNS, daysValid)
+	tlsMap[key] = p
+	return p.Cert
+}
+
+func (ctx StaticCtx) tlsKeyFromCa(caName, certName, cn string, args ...interface{}) string {
```

Comment:
same here. can you make this take explicit args if no other reason than following the convention from tlsKey? thanks

Reply:
I'm not sure I understand, this is basically following the convention from tlsKey (except dropping the legacy reference by only certName).